aiida/schedulers/plugins/slurm.py

```diff
@@ -707,53 +707,71 @@ def _parse_kill_output(self, retval, stdout, stderr):
-        lines = detailed_stdout.splitlines()
+            type_check(detailed_stdout, str)
 
-        try:
-            master = lines[1]
-        except IndexError:
-            raise ValueError('the `detailed_job_info.stdout` contained less than two lines.')
+            # The format of the detailed job info should be a multiline string, where the first line is the header, with
+            # the labels of the projected attributes. The following line should be the values of those attributes for
+            # the entire job. Any additional lines correspond to those values for any additional tasks that were run.
+            lines = detailed_stdout.splitlines()
 
-        attributes = master.split('|')
+            try:
+                master = lines[1]
+            except IndexError:
+                raise ValueError('the `detailed_job_info.stdout` contained less than two lines.')
 
-        # Pop the last element if it is empty. This happens if the `master` string just finishes with a pipe
-        if not attributes[-1]:
-            attributes.pop()
+            attributes = master.split('|')
 
-        if len(self._detailed_job_info_fields) != len(attributes):
-            raise ValueError(
-                'second line in `detailed_job_info.stdout` differs in length with schedulers `_detailed_job_info_fields'
-            )
+            # Pop the last element if it is empty. This happens if the `master` string just finishes with a pipe
+            if not attributes[-1]:
+                attributes.pop()
+
+            if len(self._detailed_job_info_fields) != len(attributes):
+                raise ValueError(
+                    'second line in `detailed_job_info.stdout` differs in length with the `_detailed_job_info_fields '
+                    'attribute of the scheduler.'
+                )
+
+            data = dict(zip(self._detailed_job_info_fields, attributes))
+
+            if data['State'] == 'OUT_OF_MEMORY':
+                return CalcJob.exit_codes.ERROR_SCHEDULER_OUT_OF_MEMORY
+
+            if data['State'] == 'TIMEOUT':
+                return CalcJob.exit_codes.ERROR_SCHEDULER_OUT_OF_WALLTIME
+
+        # Alternatively, if the ``detailed_job_info`` is not defined or hasn't already determined an error, try to match
+        # known error messages from the output written to the ``stderr`` descriptor.
+        if stderr is not None:
+
+            type_check(stderr, str)
+            stderr_lower = stderr.lower()
 
-        data = dict(zip(self._detailed_job_info_fields, attributes))
+            if re.match(r'.*exceeded.*memory limit.*', stderr_lower):
+                return CalcJob.exit_codes.ERROR_SCHEDULER_OUT_OF_MEMORY
 
-        if data['State'] == 'OUT_OF_MEMORY':
-            return CalcJob.exit_codes.ERROR_SCHEDULER_OUT_OF_MEMORY  # pylint: disable=no-member
+            if re.match(r'.*cancelled at.*due to time limit.*', stderr_lower):
+                return CalcJob.exit_codes.ERROR_SCHEDULER_OUT_OF_MEMORY
```

Comment:
Damn, just noticed this small error here.

Reply:
Eurgh, would it not have been good to add some tests in this PR, to catch things like this

Reply:
Yep; I think it's fair to do this in the follow-up PR that adds support for the remaining schedulers.
Did not want to add more work for @sphuber who jumped in quickly here

Reply:
Slow and steady wins the race 😅

Reply:
Nordic walking?

Reply:
It also shouldn't have been squashed merged. I will correct both mistakes and force push

Reply:
And I wasn't really thinking this was ready to be merged. Would have suggested to test this in the wild with SLURM to see if the regex patterns actually match. Had just taken them from the issue discussion, but not sure that these are actually correct.

Reply:
Yeh, I moaned at @sphuber about this before lol; for maintainers lets not merge each other's PRs, unless given "permission". At least I don't want people merging my PRs 😜

Reply:
Anyway, all good now. Corrected the commits on `develop`.

Reply:
> for maintainers lets not merge each other's PRs, unless given "permission" 

Ok, will keep it in mind as a policy on this repository for the future.

I typically mark my PRs as "draft" if I don't want them to be merged yet, sorry for the confusion.

Reply:
It's not that they are not ready to be merged, it's that I want to make sure that it is merged correctly (squash/merge/rebase), and that the commit message is correct